acnportal/acnsim/base.py

```diff
@@ -0,0 +1,649 @@
+"""
+This module contains a base class shared by all ACN-Sim objects.
+"""
+import json
+import operator
+import os
+# noinspection PyProtectedMember
+from pydoc import locate
+import warnings
+import pkg_resources
+import pandas
+from pandas.io.common import stringify_path, get_handle, get_filepath_or_buffer
+import numpy
+
+__NOT_SERIALIZED_FLAG__ = '__NOT_SERIALIZED__'
+
+
+# Uses methodology discussed in https://stackoverflow.com/a/16372436
+# by Martijn Pieters.
+
+
+def _operator_error_hooks(cls):
+    operator_hooks = [name for name in dir(operator)
+                      if name.startswith('__') and name.endswith('__')]
+
+    def add_hook(name):
+        # noinspection PyUnusedLocal
+        def op_hook(*args, **kwargs):
+            raise TypeError(
+                f"This object is a stub object whose methods are not "
+                f"callable. Call {name} after correctly instantiating this "
+                f"object."
+            )
+        try:
+            setattr(cls, name, op_hook)
+        except (AttributeError, TypeError):
+            pass
+
+    for hook_name in operator_hooks:
+        add_hook(hook_name)
+    return cls
+
+
+@_operator_error_hooks
+class ErrorAllWrapper:
+    """
+    This wrapper class wraps a string representing an object that acnsim
```

Comment:
I decided on a wrapper here so that we could have the desired behavior of throwing errors for most computations while still retaining access to the data in the JSON file. @zach401, does this design look good?

Reply:
We can, it's just not that useful. Every python object inherits a repr function that looks like:
"{class name} object at {memory address}". I think even if we can't serialize an object, we should still retain the class name at the very least. I figured the memory address would be useful for distinguishing unserialized objects, as these objects' ids are not recorded since they are not BaseSimObj-like objects and thus don't go in the context dict. I think it'd still be possible to write the unserializable object's id into the context dict if we want to go that route, but I like the current structure of the context dict only having BaseSimObj-likes in it.